mantidproject/mantid
# Add ReflectometryISISCalibration algorithm

**Description of work.**

This PR is the first step required for issue #32022. It adds the new calibration algorithm required to support 2D detectors. The algorithm corrects the Y positions of the detectors in the input workspace using the positions provided in a calibration file. This algorithm is still a work in progress - we have assumed that the positions of all the detectors included in the calibration file should be corrected, however we are waiting on confirmation from the scientists as to whether it should be possible to apply only a subset of the corrections from the calibration file.

At the moment the algorithm is creating a CalibTable workspace in the ADS to give us visibility over the calibration values being applied. In this PR a hard-coded output name is used, meaning that if this algorithm was run multiple times for different input workspaces you would get only one CalibTable that was overwritten by each subsequent run. It seems more sensible for the CalibTable workspace to be created with a unique name instead. As this is still a work in progress, that will be changed in a future PR as it becomes even more of a consideration once we're calling this algorithm from our existing reduction algorithms.

Support for workspace groups will be added in a future PR, if needed.

**To test:**

1) For now it's probably sufficient that the unit tests pass.
2) The reviewer could also create the workspace given in the example script on the linked issue and run the `ReflectometryISISCalibration` algorithm from Workbench using the calibration file in the example. The algorithm will not currently give the same output as running the script (it will calibrate a larger range of detectors), however it should still complete without error.

Refs #32022. 



*This does not require release notes* because release notes will be added in a future PR that implements more of the functionality required for the issue.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
